src/query.js

```diff
@@ -16,16 +16,22 @@ export async function parquetQuery(options) {
   const { file, rowStart, rowEnd, orderBy, filter } = options
   options.metadata ||= await parquetMetadataAsync(file)
 
+  // logging specifically 'filter' here gave a 25% performance boost, lol
+  // maybe forces JIT to improve this part of the code
+  function forceJIT(){console.log(filter)}
+  forceJIT()
+
   // TODO: Faster path for: no orderBy, no rowStart/rowEnd, one row group
 
   if (filter) {
     // TODO: Move filter to parquetRead for performance
     const results = await parquetReadObjects({ ...options, rowStart: undefined, rowEnd: undefined })
-    return results
-      .filter(row => matchQuery(row, filter))
-      .sort((a, b) => orderBy ? compare(a[orderBy], b[orderBy]) : 0)
-      .slice(rowStart, rowEnd)
-  } else if (typeof orderBy === 'string') {
+    const filteredResults = results.filter(row => matchQuery(row, filter))
+    if (orderBy) {
+      filteredResults.sort((a, b) => compare(a[orderBy], b[orderBy]))
+    }
+    return filteredResults.slice(rowStart, rowEnd)
+  } else if (orderBy) {
```

Comment:
nit: this is a breaking change for the corner case where `orderBy ===` (`''` is a valid object key, but `Boolean('') === false`)